deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java

```diff
@@ -121,14 +121,50 @@ public void postProcess() {
     }
 
     @Override
-    protected String getSymbolName(String input) {
-        for (Entry<String, String> entry : specialCharReplacements.entrySet()) {
-            if (input.startsWith(entry.getKey())) {
-                return input.length() > entry.getKey().length()
-                        ? entry.getValue() + "_" + input.substring(entry.getKey().length())
-                        : entry.getValue() + "_symbol";
+    public String toEnumVarName(String value, String datatype) {
+        String enumValue = value;
+        if (enumValue.isEmpty()) {
+            return "EMPTY";
+        }
+
+        if (getSymbolName(enumValue) != null) {
+            return getSymbolName(enumValue).toUpperCase(Locale.ROOT);
+        }
+
+        if (" ".equals(enumValue)) {
+            return "SPACE";
+        }
+
+        if ("Integer".equals(datatype) || "Long".equals(datatype) ||
+                "Float".equals(datatype) || "Double".equals(datatype) || "BigDecimal".equals(datatype)) {
+            String varName = "NUMBER_" + enumValue;
+            varName = varName.replaceAll("-", "MINUS_");
+            varName = varName.replaceAll("\\+", "PLUS_");
+            varName = varName.replaceAll("\\.", "_DOT_");
+            return varName;
+        }
+
+        for (Map.Entry<String, String> entry : this.specialCharReplacements.entrySet()) {
+            if (enumValue.contains(entry.getKey())) {
+                enumValue = enumValue.replace(entry.getKey(), "_".concat(entry.getValue()).concat("_"));
             }
         }
-        return null;
+
+        return enumValue
```

Comment:
This is an alternative implementation of super method section, which is the "effect" we are trying to achieve? 
```
// string
        String var = value.replaceAll("\\W+", "_").toUpperCase(Locale.ROOT);
        if (var.matches("\\d.*")) {
            return "_" + var;
        } else {
            return var;
        }
```